src/graph.py

```diff
@@ -30,65 +30,114 @@ def __init__(self):
 
     def add_node(self, node):
         """Add a node to the graph."""
-        if node in self.graph.keys():
+        if node in list(self.graph.keys()):
             raise KeyError("Node already in graph.")
-        self.graph[node] = {"edges": []}
+        self.graph[node] = {}
 
-    def add_edge(self, node1, node2):
+    def add_edge(self, node1, node2, weight=0):
         """Add an edge from node1 to node2 in the graph."""
-        if node1 not in self.graph.keys():
+        if node1 not in list(self.graph.keys()):
             self.add_node(node1)
-        if node2 not in self.graph.keys():
+        if node2 not in list(self.graph.keys()):
             self.add_node(node2)
-        if node2 not in self.graph[node1]['edges']:
-            self.graph[node1]['edges'].append(node2)
+        if node2 not in list(self.graph[node1].keys()):
+            self.graph[node1][node2] = weight
 
     def nodes(self):
         """Return a list of all nodes in the graph."""
         node_list = []
-        for node in self.graph.keys():
+        for node in list(self.graph.keys()):
             node_list.append(node)
         return node_list
```

Comment:
We added list() around self.graph.keys() because of py2/ py3 compatibility issues. Is there a better fix for that?